subprojects/maven/src/main/java/org/gradle/api/publish/maven/MavenDependency.java

```diff
@@ -21,7 +21,10 @@
 
 /**
  * A dependency declared as part of an {@link MavenPublication}.
+ *
+ * @deprecated This type is not referenced by any other public API classes. It will be removed in Gradle 9.0
```

Comment:
We never returned instances of this type to users in any API. It is essentially a "free floating" interface with no instances that implement (from the user's perspective)

Reply:
I think you should deprecate the whole type hierarchy as well and log in the constructors of the default implementations, even if they aren't public API, just in case.  There's still a chance a plugin author is accessing them via internal API or something.

Reply:
Discussed during standup. Nothing to do here.